runelite-client/src/main/java/net/runelite/client/plugins/cluescrolls/ClueScrollPlugin.java

```diff
@@ -217,11 +220,31 @@ public void onGameTick(final GameTick event)
 
 			if (objectId != -1)
 			{
-				GameObjectQuery query = new GameObjectQuery().idEquals(objectId);
-				objectsToMark = queryRunner.runQuery(query);
+				if (clue instanceof LocationClueScroll)
+				{
+					// Match object with location every time
+					final WorldPoint location = ((LocationClueScroll) clue).getLocation();
+					LocalPoint localLocation = LocalPoint.fromWorld(client, location);
+
+					if (localLocation != null)
+					{
+						final Region region = client.getRegion();
+						final Tile[][][] tiles = region.getTiles();
+						final Tile tile = tiles[client.getPlane()][localLocation.getRegionX()][localLocation.getRegionY()];
+
+						objectsToMark = Arrays.stream(tile.getGameObjects())
+							.filter(object -> object != null && object.getId() == objectId)
+							.toArray(GameObject[]::new);
+					}
+				}
+				else
+				{
+					// All data we have is object id, so try to at least mark something
+					objectsToMark = queryRunner.runQuery(new GameObjectQuery().idEquals(objectId));
```

Comment:
There are no ObjectClueScrolls that aren't also LocationClueScrolls, so this else has no use.

Reply:
Yeah, I know, but it is just in case.

Reply:
Just in case Jagex adds another clue type? Umm...

Reply:
Just in case someone will forget/dont have data about location, only the object ID for w/e reason. It's mostly just so we will see at least something and not just wonder why clue is not highlighting anything in future. Idk, it do not hurts to have it, even though the scenario where it will apply is not that realistic.